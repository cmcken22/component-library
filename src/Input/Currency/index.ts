// Generated with util/create-component.js
export { CurrencyInputProps, default } from "./Currency";
