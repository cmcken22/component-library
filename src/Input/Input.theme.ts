export default {
  styleOverrides: {
    root: ({ theme, ownerState }: any) => {
      return {};
    },
  },
};
