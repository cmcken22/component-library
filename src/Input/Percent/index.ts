// Generated with util/create-component.js
export { PercentInputProps, default } from "./Percent";
