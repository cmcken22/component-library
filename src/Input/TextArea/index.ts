// Generated with util/create-component.js
export { TextAreaProps, default } from "./TextArea";
