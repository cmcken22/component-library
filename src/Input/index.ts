export { InputProps, default } from "./Input";
