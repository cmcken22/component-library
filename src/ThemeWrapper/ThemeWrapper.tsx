import React from "react";
import { StyledEngineProvider, ThemeProvider, CssBaseline } from "@mui/material";
import { theme } from "../theme";
import '../index.css';

const ThemeWrapper = ({ children }: any) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

export default ThemeWrapper;
