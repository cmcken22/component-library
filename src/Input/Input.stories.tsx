import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import Input from "./Input";

export default {
  title: "ReactComponentLibrary/Input",
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;
const CurrencyTemplate: ComponentStory<typeof Input> = (args) => (
  <Input.Currency {...args} />
);
const PercentTemplate: ComponentStory<typeof Input> = (args) => (
  <Input.Percent {...args} />
);
const TextAreaTemplate: ComponentStory<typeof Input> = (args) => (
  <Input.TextArea {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  id: "Primary",
  label: "Primary",
  placeholder: "Enter a value",
  disabled: false,
  helperText: "Helper Text",
  required: false,
  value: "Hello world",
  fullWidth: false,
  labelPosition: "top",
  onChange: (e) => action("onChange")(e),
};

Primary.argTypes = {
  status: {
    options: ["error", "success", "warning"],
    control: { type: "select" },
  },
  labelPosition: {
    options: ["top", "left"],
    control: { type: "select" },
  },
};

export const Currency = CurrencyTemplate.bind({});
Currency.args = {
  id: "Currency",
  label: "Currency",
  placeholder: "Enter a value",
  disabled: false,
  helperText: "Helper Text",
  required: false,
  value: "100",
  fullWidth: false,
  labelPosition: "top",
  onChange: (value, formattedValue, floatValue) =>
    action("onChange")(value, formattedValue, floatValue),
};

Currency.argTypes = {
  status: {
    options: ["error", "success", "warning"],
    control: { type: "select" },
  },
  labelPosition: {
    options: ["top", "left"],
    control: { type: "select" },
  },
};

export const Percent = PercentTemplate.bind({});
Percent.args = {
  id: "Percent",
  label: "Percent",
  placeholder: "Enter a value",
  disabled: false,
  helperText: "Helper Text",
  required: false,
  value: "100",
  fullWidth: false,
  labelPosition: "top",
  onChange: (value, formattedValue, floatValue) =>
    action("onChange")(value, formattedValue, floatValue),
};

Percent.argTypes = {
  status: {
    options: ["error", "success", "warning"],
    control: { type: "select" },
  },
  labelPosition: {
    options: ["top", "left"],
    control: { type: "select" },
  },
};

export const TextArea = TextAreaTemplate.bind({});
TextArea.args = {
  id: "TextArea",
  label: "Title",
  placeholder: "Type here...",
  value:
    "I’m typing a bunch of text here just to show what it looks like when it’s filled.",
  required: true,
  onChange: (e) => action("onChange")(e),
};
