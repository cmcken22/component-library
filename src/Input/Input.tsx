import { TextField } from "@mui/material";
import cx from "classnames";
import React, { useCallback, useState } from "react";
import BaseInput, { BaseInputProps } from "../BaseInput";
import Currency from "./Currency";
import Percent from "./Percent";
import TextArea from "./TextArea";

// export interface BaseInputProps {
//   id?: string;
//   status?: "error" | "warning" | "success";
//   disabled?: boolean;
//   fullWidth?: boolean;
//   children?:
//     | React.ReactNode
//     | React.JSX.Element
//     | ((props: any) => React.ReactNode);
//   containerSx?: any;
// }

export interface InputProps extends BaseInputProps {
  label?: string;
  placeholder?: string;
  helperText?: string;
  value?: string;
  required?: boolean;
  labelPosition?: "top" | "left";
  onChange?: (value: string) => void;
  endAdornment?: React.ReactNode;
  fullWidth?: boolean;
  disabled?: boolean;
  // multiline?: boolean;
  // maxRows?: number;
  // minRows?: number;
  // type?: string;
}

const Input = ({
  id,
  label,
  placeholder,
  status,
  helperText,
  disabled,
  value: passedValue,
  fullWidth,
  required,
  labelPosition = "top",
  onChange,
}: InputProps) => {
  const [value, setValue] = useState(passedValue || "");

  const handleChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setValue(e.target.value);
      if (onChange) onChange(e.target.value);
    },
    [onChange, setValue]
  );

  return (
    <BaseInput id={id} status={status}>
      {({ endAdornment, className }: any) => (
        <>
          <BaseInput.Label required={required} position={labelPosition}>
            {label}
          </BaseInput.Label>
          <TextField
            className={cx(className)}
            placeholder={placeholder}
            value={value}
            variant="outlined"
            disabled={disabled}
            fullWidth={fullWidth}
            onChange={handleChange}
            InputProps={{ endAdornment }}
          />
          <BaseInput.HelperText>{helperText}</BaseInput.HelperText>
        </>
      )}
    </BaseInput>
  );
};

Input.defaultProps = {
  labelPosition: "top",
  multiline: false,
};

Input.Currency = Currency;
Input.Percent = Percent;
Input.TextArea = TextArea;

export default Input;
