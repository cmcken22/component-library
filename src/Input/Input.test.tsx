import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import "jest-styled-components";
import React from "react";
import ThemeWrapper from "../ThemeWrapper";
import Input, { InputProps } from "./Input";

// let documentBody: RenderResult;

const AllTheProviders = ({ children }) => {
  return <ThemeWrapper>{children}</ThemeWrapper>;
};

const customRender = (ui, options) =>
  // render(ui, { wrapper: AllTheProviders, ...options });
  render(ui, { wrapper: AllTheProviders, ...options });

describe("Test Component", () => {
  let props: InputProps;

  beforeEach(() => {
    // documentBody = render(provideTheme(<Input />));
    // document.head.innerHTML = document.head.innerHTML; // ⚠️ Add this line.
    // console.log("InputComponent:", document.head.innerHTML);
    props = {
      id: "test_id",
      label: "label",
      placeholder: "placeholder",
      helperText: "helperText",
      disabled: false,
      value: "12345",
      fullWidth: false,
      required: true,
      labelPosition: "top",
    };
  });

  const renderComponent = () => {
    const component = customRender(<Input {...props} />, {});
    return component;
  };

  it("should render correctly", () => {
    const { getByTestId } = renderComponent();
    const component = getByTestId("Input");
    expect(component).toBeInTheDocument();

    const InputComponent = component.querySelector("input");
    expect(InputComponent).toBeInTheDocument();

    expect(component).toHaveAttribute("id", "test_id");
    expect(InputComponent).toHaveValue("12345");
  });

  it("should update value", () => {
    const { getByTestId } = renderComponent();
    const component = getByTestId("Input");

    const Input = component.querySelector("input");
    expect(Input).toBeInTheDocument();

    // Simulate a change event
    fireEvent.change(Input!, { target: { value: "5678" } });

    // Check if the value has been updated
    expect(Input).toHaveValue("5678");
  });

  it("should have error state", () => {
    props.status = "error";
    // console.log("body:", document.body.innerHTML);
    const { getByTestId } = renderComponent();
    // console.log("head:", document.head.outerHTML);
    const component = getByTestId("Input");
    const InputComponent = component.querySelector(".MuiTextField-root");
    // console.log("InputComponent:", document.head.innerHTML);
    // setTimeout(() => {
    // const FieldSet = component.querySelector(".MuiOutlinedInput-root fieldset");
    // const computedStyle = document.defaultView.getComputedStyle(FieldSet);
    // console.log("computedStyle", computedStyle?.borderColor);
    // }, 1000);
    // console.log("FieldSet:", getStyle(FieldSet, "border-bottom-color"));
    // expect(InputComponent).toBeInTheDocument();
    expect(InputComponent).toHaveClass("status--error");
  });
});
