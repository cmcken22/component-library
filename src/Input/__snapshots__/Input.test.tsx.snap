// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly: Input test 1 1`] = `
<div>
  <div
    class="MuiBox-root css-tot46x"
    data-testid="Input"
    id="test_id"
  >
    <div
      class="Input-container MuiBox-root css-eye1ln"
    >
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-fieldLabel Input__label css-tl9j0x-MuiTypography-root"
        >
          This is a label
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-rh3oz2-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1m54fg7-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-smitoi-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="placeholder"
            type="text"
            value="12345"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-errorMessage Input__helper-text css-3yxls5-MuiTypography-root"
      >
        helperText
      </span>
    </div>
  </div>
</div>
`;

exports[`Input should have error state: Input error state 1`] = `
<div>
  <div
    class="MuiBox-root css-v8lsnt"
    data-testid="Input"
    id="test_id"
  >
    <div
      class="Input-container MuiBox-root css-eye1ln"
    >
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-fieldLabel Input__label css-tl9j0x-MuiTypography-root"
        >
          This is a label
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root status--error css-rh3oz2-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1tjabls-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-oq1q8e-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r7:"
            placeholder="placeholder"
            type="text"
            value="12345"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="icon-wrapper undefined MuiBox-root css-nuxmg4"
            >
              <svg
                fill="transparent"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.1712 14.8284L11.9996 12M14.8281 9.17157L11.9996 12M11.9996 12L9.1712 9.17157M11.9996 12L14.8281 14.8284"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-errorMessage Input__helper-text css-3yxls5-MuiTypography-root"
      >
        helperText
      </span>
    </div>
  </div>
</div>
`;

exports[`Input should have success state: Input success state 1`] = `
<div>
  <div
    class="MuiBox-root css-us1v1j"
    data-testid="Input"
    id="test_id"
  >
    <div
      class="Input-container MuiBox-root css-eye1ln"
    >
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-fieldLabel Input__label css-tl9j0x-MuiTypography-root"
        >
          This is a label
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root status--success css-rh3oz2-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1tjabls-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-oq1q8e-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r5:"
            placeholder="placeholder"
            type="text"
            value="12345"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="icon-wrapper undefined MuiBox-root css-nuxmg4"
            >
              <svg
                fill="transparent"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 12.5L10 15.5L17 8.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-errorMessage Input__helper-text css-3yxls5-MuiTypography-root"
      >
        helperText
      </span>
    </div>
  </div>
</div>
`;

exports[`Input should have warning state: Input warning state 1`] = `
<div>
  <div
    class="MuiBox-root css-1unn91"
    data-testid="Input"
    id="test_id"
  >
    <div
      class="Input-container MuiBox-root css-eye1ln"
    >
      <div
        class="MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-fieldLabel Input__label css-tl9j0x-MuiTypography-root"
        >
          This is a label
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root status--warning css-rh3oz2-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1tjabls-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-oq1q8e-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r6:"
            placeholder="placeholder"
            type="text"
            value="12345"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="icon-wrapper undefined MuiBox-root css-nuxmg4"
            >
              <svg
                fill="transparent"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.0429 21H3.95705C2.41902 21 1.45658 19.3364 2.22324 18.0031L10.2662 4.01533C11.0352 2.67792 12.9648 2.67791 13.7338 4.01532L21.7768 18.0031C22.5434 19.3364 21.581 21 20.0429 21Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M12 9V13"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M12 17.01L12.01 16.9989"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-errorMessage Input__helper-text css-3yxls5-MuiTypography-root"
      >
        helperText
      </span>
    </div>
  </div>
</div>
`;
